src/dataflow/src/render/mod.rs

```diff
@@ -320,7 +329,7 @@ pub(crate) fn build_dataflow<A: Allocate>(
                                         &dataflow.debug_name,
                                         worker_index,
                                         as_of_frontier.clone(),
-                                        &mut src.operators,
+                                        &src.operators,
```

Comment:
Fwiw, the intent of passing a `&mut` reference in is so that the method can communicate whether it has applied the operator or not. For example, I think this method always plans to apply it, and so can replace it with `None` once it has done so. That will prevent a repeated application just a bit down (I think) from here.